src/modm/ui/color/rgb.hpp

```diff
@@ -0,0 +1,190 @@
+	{}
+
+	TSum
+	getColorSum() const
+	{
+		return red + green + blue;
+	}
+
+	template<typename IntermediateType = float, std::unsigned_integral ReturnType = T>
+			requires std::is_fundamental_v<IntermediateType>
+	ReturnType
+	getRelative(const T color, IntermediateType multiplier = 100) const
+	{
+		return IntermediateType(color) * multiplier / IntermediateType(getColorSum());
+	}
+
+	template<typename IntermediateType = float, std::unsigned_integral ReturnType = T>
+			requires std::is_fundamental_v<IntermediateType>
+	ReturnType
+	getRelativeRed(IntermediateType multiplier = 100) const
+	{
+		return getRelative<IntermediateType, ReturnType>(red, multiplier);
+	}
+
+	template<typename IntermediateType = float, std::unsigned_integral ReturnType = T>
+			requires std::is_fundamental_v<IntermediateType>
+	ReturnType
+	getRelativeGreen(IntermediateType multiplier = 100) const
+	{
+		return getRelative<IntermediateType, ReturnType>(green, multiplier);
+	}
+
+	template<typename IntermediateType = float, std::unsigned_integral ReturnType = T>
+			requires std::is_fundamental_v<IntermediateType>
+	ReturnType
+	getRelativeBlue(IntermediateType multiplier = 100) const
+	{
+		return getRelative<IntermediateType, ReturnType>(blue, multiplier);
+	}
+
+	template<typename IntermediateType = float, std::unsigned_integral ReturnType = T>
+			requires std::is_fundamental_v<IntermediateType>
+	RgbT<ReturnType>
+	getRelativeColors(IntermediateType multiplier = 100) const
+	{
+		return RgbT<ReturnType>(getRelativeRed<IntermediateType, ReturnType>(multiplier),
+								getRelativeGreen<IntermediateType, ReturnType>(multiplier),
+								getRelativeBlue<IntermediateType, ReturnType>(multiplier));
+	}
+
+	/**
+	 * Normalize color values based on a clear value
+	 *
+	 * Imagine a low band light, for example a green laser. In case the filters
+	 * of a sensor do not transfer this wavelength well, it might result in all
+	 * colors being very low. An aditional clear value (provided f.e. by TCS3472)
+	 * will not filter colors and thus it will see a bright light (intensity).
+	 * In order to still have some signal the very low green value can be
+	 * amplified with the clear value.
+	 */
+	template<typename IntermediateType = float>
+			requires std::is_fundamental_v<IntermediateType>
+	void
+	normalizeByValue(T value)
+	{
+		const IntermediateType scalar = value / IntermediateType(getColorSum());
+		red *= scalar;
+		green *= scalar;
+		blue *= scalar;
+	}
+
+	constexpr bool
+	operator==(const RgbT<T>& other) const = default;
+
+private:
+	template<std::unsigned_integral U>
+	friend IOStream&
+	operator<<(IOStream& os, const RgbT<U>&);
+};
+
+/// @ingroup modm_ui_color
+using Rgb = RgbT<uint8_t>;
+
+
+#if __has_include(<modm/io/iostream.hpp>)
+#include <modm/io/iostream.hpp>
+
+template<std::unsigned_integral U>
+IOStream&
+operator<<(IOStream& os, const color::RgbT<U>& color)
+{
+	os << color.red << "\t" << color.green << "\t" << color.blue;
+	return os;
+}
+#endif
```

Comment:
We typically don't depend on the `:io` module and instead guard the IO serialization implementation this way.
This keeps the `:io` module optional, since it pulls in *lot* code…